lib/src/drag_select_grid_view/drag_select_grid_view.dart

```diff
@@ -208,16 +206,17 @@ class DragSelectGridViewState extends State<DragSelectGridView>
   @override
   void handleScroll() {
     if (_lastMoveUpdateDetails != null) {
-      _handleLongPressMoveUpdate(_lastMoveUpdateDetails);
+      _handleLongPressMoveUpdate(_lastMoveUpdateDetails!);
     }
```

Comment:
I don't understand what this would improve. `_lastMoveUpdateDetails` can not be null here so it is safe to add the `!` operator.

Reply:
Actually, `_lastMoveUpdateDetails` can be null. It is quite common to think that this behavior is a limitation in Dart's type propagation, but it is not. By calling `_lastMoveUpdateDetails` again, nothing assures it'll give you the same result as before. Suppose `_lastMoveUpdateDetails` was a getter:

```dart
int count = 0;
LongPressMoveUpdateDetails? _backingField;
LongPressMoveUpdateDetails? get _lastMoveUpdateDetails => count++ == 0 ? _backingField: null;

@override
void handleScroll() {
  if (_lastMoveUpdateDetails != null) { // first call returns a proper value.
    _handleLongPressMoveUpdate(_lastMoveUpdateDetails!); // second call returns null.
  }
}
```

We are not working with a getter, but the field could also be overriden:

```dart
class State {
  LongPressMoveUpdateDetails? _lastMoveUpdateDetails;

  @override
  void handleScroll() {
    if (_lastMoveUpdateDetails != null) { // first call returns a proper value.
      _handleLongPressMoveUpdate(_lastMoveUpdateDetails!); // second call returns null.
    }
  }
  
  ...
}

class MyState extends State {
  int count = 0;
  LongPressMoveUpdateDetails? _lastMoveUpdateDetailsBackingField;

  @override
  LongPressMoveUpdateDetails? get _lastMoveUpdateDetails =>
    count++ == 0 ? _lastMoveUpdateDetailsBackingField : null;
}

void main() {
  State().handleScroll(); // works fine.
  MyState().handleScroll(); // throws.
}
```

These problems are even valid for final fields, since their getter can also be overriden. The only possible way to work around this is by not making multiple calls. That's the reason why you have to assign the result to a variable.

Reply:
Hmm indeed I didn't think of this edge case. I do not have an opinion yet on how to best handle this and might not find the time today to think this through. Please feel free to just decide how you want to do proceed. You can just push to my branch.

Reply:
This sort of problem started to happen a lot after nnbd, but it is quite old and very recurrent. We (the global community) have been solving it the way I described. As I told you, there's no escaping. Even final fields share the issue.

Different languages solve this differently. In Kotlin, classes are final by default, so final fields are safe. When the field is not final or the class is not final, we resort to scoped functions.